### PR TITLE
Week6

### PR DESCRIPTION
**Pull request by DzmitryU**
Added server requests and loading

### CODE REVIEW

File: src/ac/index.js [hunk 1]
```diff
@@ -1,15 +1,24 @@
-import {LOAD_ALL_ARTICLES, SELECT_ARTICLES, SET_DATE_RANGE, LOAD_COMMENTS, INCREMENT, ADD_COMMENT} from '../constants';
+import {
+    LOAD_ARTICLES, SELECT_ARTICLES, SET_DATE_RANGE, LOAD_COMMENTS, INCREMENT, ADD_COMMENT, DELETE_ARTICLE
```
Comment (sajdre) [inline src/ac/index.js]: why did you delete load all articles action?

File: src/components/CommentList.js [hunk 2]
```diff
@@ -26,25 +33,49 @@ function getBody(comments, isOpen, articleId) {
...
-        </div>
-    );
+class CommentList extends React.Component {
+    constructor(props) {
+        super(props);
+    };
 
+    handleClick = () => {
+        if (!(this.props.isOpen || this.props.loading || this.props.loaded)) {
+            this.props.loadComments(this.props.articleId);
```
Comment (sajdre) [inline src/components/CommentList.js]: it's bad practice to load comments by click, imagine you need to load comments if button doesn't exist, e.g. on mobile devices, so maybe move loading to ComponentWillReceiveProps is better idea, what do you think?

File: src/middlewares/api.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import {SUCCESS, START} from '../constants'
+
+export default store => next => action => {
```
Comment (sajdre) [inline src/middlewares/api.js:3]: let's use thunk

**Review by sajdre — CHANGES_REQUESTED** [review]

File: src/ac/index.js (hunk 1 above)
Comment (DzmitryU) [reply src/ac/index.js]: It's replaced by 'load articles' action

File: src/middlewares/api.js (hunk 3 above)
Comment (DzmitryU) [reply src/middlewares/api.js:3]: Done

**Pull request merged**